compiler/src/dotty/tools/dotc/transform/PickleQuotes.scala

```diff
@@ -76,375 +83,80 @@ class PickleQuotes extends MacroTransform {
-     *  are in the body of an inline method.
-     */
-     protected def transformSpliceType(body: Tree, splice: Select)(using Context): Tree =
-      if level > 1 then
-        val body1 = nested(isQuote = false).transform(body)(using spliceContext)
-        cpy.Select(splice)(body1, splice.name)
-      else if level == 1 then
-        val (body1, quotes) = nested(isQuote = false).splitSplice(body)(using spliceContext)
-        val tpe = outer.embedded.getHoleType(body, splice)
-        makeHole(splice.isTerm, body1, quotes, tpe).withSpan(splice.span)
-      else
-        splice
-
-    /** Transforms the contents of a nested splice
-     *  Assuming
-     *     '{
-     *        val x = ???
-     *        val y = ???
-     *        ${ ... '{ ... x .. y ... } ... }
-     *      }
-     *  then the spliced subexpression
-     *     { ... '{ ... x ... y ... } ... }
-     *  will be transformed to
-     *     (args: Seq[Any]) => {
-     *       val x$1 = args(0).asInstanceOf[Expr[Any]] // or .asInstanceOf[Type[Any]]
-     *       val y$1 = args(1).asInstanceOf[Expr[Any]] // or .asInstanceOf[Type[Any]]
-     *       { ... '{ ... ${x$1} ... ${y$1} ... } ... }
-     *     }
-     *
-     *  See: `capture`
-     *
-     *  At the same time register embedded trees `x` and `y` to place as arguments of the hole
-     *  placed in the original code.
-     *     '{
-     *        val x = ???
-     *        val y = ???
-     *        Hole(0 | x, y)
-     *      }
-     */
-    private def makeLambda(tree: Tree)(using Context): Tree = {
-      def body(arg: Tree)(using Context): Tree = {
-        var i = 0
-        transformWithCapturer(tree)(
-          (captured: mutable.Map[Symbol, Tree]) => {
-            (tree: Tree) => {
-              def newCapture = {
-                val tpw = tree.tpe.widen match {
-                  case tpw: MethodicType => tpw.toFunctionType(isJava = false)
-                  case tpw => tpw
-                }
-                assert(tpw.isInstanceOf[ValueType])
-                val argTpe =
-                  if (tree.isType) defn.QuotedTypeClass.typeRef.appliedTo(tpw)
-                  else defn.FunctionType(1, isContextual = true).appliedTo(defn.QuotesClass.typeRef, defn.QuotedExprClass.typeRef.appliedTo(tpw))
-                val selectArg = arg.select(nme.apply).appliedTo(Literal(Constant(i))).cast(argTpe)
-                val capturedArg = SyntheticValDef(UniqueName.fresh(tree.symbol.name.toTermName).toTermName, selectArg)
-                i += 1
-                embedded.addTree(tree, capturedArg.symbol)
-                captured.put(tree.symbol, capturedArg)
-                capturedArg
-              }
-              val refSym = captured.getOrElseUpdate(tree.symbol, newCapture).symbol
-              ref(refSym).withSpan(tree.span)
+  private def makeHoles(tree: tpd.Tree)(using Context): (List[Tree], tpd.Tree) =
+
+    class HoleContentExtractor extends Transformer:
+      private var contents = List.newBuilder[Tree]
```

Comment:
If it's a mutable builder it probaly doesn't need to be a var too.

Reply:
Should be a `val`. Fixed.